### DISCUSSION TITLE
Create a framework for plugin creation

### DISCUSSION THREAD

**Discussion by eisraeli** [jdx/mise#2026] [Ideas]
1.  I would like to see this tool moving away from the ASDF plugin system in which each developer had to implement their own logic from scratch. All plugins implement logic of common actions like `download`, `install`, `verify`, etc.
Some of these actions share a common logic and can utilize shared code between them.

2.  I would also like to see plugins which are not written in Bash and have proper testing.
So far - community-wise - it seems like the recommended way is to create plugins in Bash while the core plugins are written in Rust. 
Can you at least add support for community-driven plugins in Rust and Go ?

**Comment by jdx** [comment]
I'm experimentally planning to adopt vfox plugins

**Comment by airtonix** [comment]
just because asdf uses bash, doesn't mean it can't be unit/integration tested.

[bats](https://github.com/bats-core/bats-core) is a thing.
